themes/default/source/js/popup.js

```diff
@@ -92,6 +98,22 @@ export default function(gallerySelector, mapArr) {
     }
   };
 
+  /**
+   * If map or deepzoom leaflet instances are defined, removes them.
+   * There should only be one defined at a time
+   * @todo The Map and Deepzoom classes should be refactored, at least to have the same signature
+   */
+  const tearDownMap = () => {
+    if (deepzoom) {
+      deepzoom.map.remove();
+      return;
+    }
+    if (map) {
+      map.remove();
+      return;
+    }
+  };
```

Comment:
We definitely need to refactor this; if the two have completely different signatures we should look at using a factory pattern to configure and instantiate a single instance of either _component_ and then an adapter module to abstract the differences.